README.md

```diff
@@ -965,112 +1408,535 @@ export function trap(message: string): Query<boolean> {
 }
 ```
 
-### Cross-canister calls
+### Call APIs
+
+-   [caller](#caller)
+-   [accept message](#accept-message)
+-   [arg data](#arg-data)
+-   [arg data raw](#arg-data-raw)
+-   [arg data raw size](#arg-data-raw-size)
+-   [call](#call)
+-   [call raw](#call-raw)
+-   [call raw 128](#call-raw-128)
+-   [call with payment](#call-with-payment)
+-   [call with payment 128](#call-with-payment-128)
+-   [method name](#method-name)
+-   [msg cycles accept](#msg-cycles-accept)
+-   [msg cycles accept 128](#msg-cycles-accept-128)
+-   [msg cycles available](#msg-cycles-available)
+-   [msg cycles available 128](#msg-cycles-available-128)
+-   [msg cycles refunded](#msg-cycles-refunded)
+-   [msg cycles refunded 128](#msg-cycles-refunded-128)
+-   [notify](#notify)
+-   [notify raw](#notify-raw)
+-   [notify with payment 128](#notify-with-payment-128)
+-   [performance counter](#performance-counter)
+-   [reject](#reject)
+-   [reject code](#reject-code)
+-   [reject message](#reject-message)
+-   [reply](#reply)
+-   [reply raw](#reply-raw)
+-   [result](#result)
+
+#### caller
+
+Examples:
+
+-   [basic-dao](/examples/motoko_examples/basic-dao)
+-   [defi](/examples/motoko_examples/defi)
+-   [ic_api](/examples/ic_api)
+-   [whoami](/examples/motoko_examples/whoami)
+
+```typescript
+import { ic, Principal, Query } from 'azle';
+
+// returns the principal of the identity that called this function
+export function caller(): Query<Principal> {
+    return ic.caller();
+}
+```
+
+#### accept message
+
+Examples:
+
+-   [inspect_message](/examples/inspect_message)
+
+#### arg data
+
+Not yet implemented.
```

Comment:
Let's link this to the related issue (496). I've update the issue to include a task for updating this readme when it's implemented.